src/hooks/useRanking.ts

```diff
@@ -1,98 +1,77 @@
-  }
-
-  const findCurrentSeason = () => {
-    for (let i = 0; i < 10; i ++) {
-        const seasonDate = seasonsDate.get(i);
-        if (seasonDate &&  
-          seasonDate.diffNow("days").days <= 0 &&
-          seasonDate.diffNow("days").days > -30) {
-                return {seasonNumber: i, seasonEndDate: seasonsDate.get(i+1)};
-            }
-    }
+  for (let i = 0; i <= 10; i++) {
+    seasonsDate.set(i, startOfSeasons.plus({ months: duration * i }));
   }
 
   const claimStatic = useAsyncFunction(
     async () => {
-      const season =  findCurrentSeason();
-      if(season && wallet.signer){
-        const result = {status:"SUCCESS"}
-        await new Promise(r => setTimeout(r, 2000));
+      if (season && wallet.signer) {
+        const result = { status: 'SUCCESS' };
+        await new Promise((r) => setTimeout(r, 2000));
         //await redeemBadge("", wallet.signer, "S"+season.seasonNumber.toString(), true);
-        return result.status === "SUCCESS";
+        return result.status === 'SUCCESS';
       }
       return false;
     },
-    useMemo(() => undefined, [])
+    useMemo(() => undefined, []),
   );
 
   const claim = useAsyncFunction(
     async () => {
-      const season =  findCurrentSeason();
-      if(season && wallet.signer){
-        const result = {status:"SUCCESS"}
-        await new Promise(r => setTimeout(r, 2000));
+      if (season && wallet.signer) {
+        const result = { status: 'SUCCESS' };
+        await new Promise((r) => setTimeout(r, 2000));
         //await redeemBadge("", wallet.signer, "S"+season.seasonNumber.toString(), false);
-        return result.status === "SUCCESS";
+        return result.status === 'SUCCESS';
       }
       return false;
     },
-    useMemo(() => undefined, [])
+    useMemo(() => undefined, []),
   );
 
-
-  const curretSeason = findCurrentSeason();
-  const seasonEndDate = curretSeason ? seasonsDate.get(curretSeason.seasonNumber) : undefined;
+  const seasonEndDate = season ? seasonsDate.get(season.seasonNumber) : undefined;
 
   const rankings = useAsyncFunction(
     async () => {
-      const season = findCurrentSeason();
       const number = season?.seasonNumber;
       const startTime = !isUndefined(number) ? seasonsDate.get(number) : undefined;
       if (startTime) {
-        const result = await getActivity({from: startTime.toSeconds()});
+        const result = await getActivity({ from: startTime.toSeconds() });
         return result.total;
       }
     },
-    useMemo(() => undefined, [])
+    useMemo(() => undefined, []),
   );
 
-  return useMemo(() => ({
-    rankings,
-    claimStatic,
-    claim,
-    seasonEndDate,
-    findCurrentSeason
-  } as useRankingReturnType),
-    [
-      rankings,
-      claimStatic,
-      claim,
-      seasonEndDate,
-      findCurrentSeason
-    ]);
-}
+  return useMemo(
+    () =>
+      ({
+        rankings,
+        claimStatic,
+        claim,
```

Comment:
does the leaderboard have support for claiming? it is not supposed to at this stage since SBTs will roll out in phase 2

Reply:
fixed in: 075cc4da9fa1be34e1cb2de394276557873761ca